canopeum_frontend/src/components/analytics/site-modal/SiteModal.tsx

```diff
@@ -84,23 +88,31 @@ const SiteModal = ({ open, handleClose, siteId }: Props) => {
     setSiteImageURL(URL.createObjectURL(blob))
   }, [siteId, getApiClient])
 
-  const fetchSiteTypes = useCallback(
-    async () => setAvailableSiteTypes(await getApiClient().siteClient.types()),
-    [getApiClient],
-  )
-
   const onImageUpload = (file: File) => {
     setSite(value => ({ ...value, siteImage: file }))
     setSiteImageURL(URL.createObjectURL(file))
   }
 
-  useEffect(() => void fetchSiteTypes(), [fetchSiteTypes])
+  useEffect(() => {
+    const fetchSiteTypes = async () =>
+      setAvailableSiteTypes(await getApiClient().siteClient.types())
+
+    fetchSiteTypes().catch((error: unknown) =>
```

Comment:
Can be simplified.

```suggestion
  useEffect(() => {
    setAvailableSiteTypes(await getApiClient().siteClient.types())().catch((error: unknown) =>
```

Actually even better, we could chain a .then, .catch to handle generic exceptions from either